src/backend/OpenCL/CHIPBackendOpenCL.cc

```diff
@@ -597,13 +595,19 @@ CHIPBackendOpenCL::createCHIPEvent(chipstar::Context *ChipCtx,
   return std::shared_ptr<chipstar::Event>(Event);
 }
 
-void CHIPEventOpenCL::recordStream(chipstar::Queue *ChipQueue) {
-  logTrace("chipstar::Event::recordStream()");
-  std::shared_ptr<chipstar::Event> MarkerEvent = ChipQueue->enqueueMarker();
-  this->takeOver(MarkerEvent);
+void CHIPQueueOpenCL::recordEvent(chipstar::Event *ChipEvent) {
+  logTrace("chipstar::Queue::recordEvent({})", (void *)ChipEvent);
+  auto ChipEventCL = static_cast<CHIPEventOpenCL *>(ChipEvent);
 
-  this->EventStatus_ = EVENT_STATUS_RECORDING;
-  return;
+  if (getLastEvent() == nullptr) {
+    std::shared_ptr<chipstar::Event> MarkerEvent = enqueueMarker();
+    ChipEventCL->takeOver(MarkerEvent);
+    ChipEventCL->setRecording();
+  } else {
+    std::shared_ptr<chipstar::Event> LastEvent = getLastEvent();
+    ChipEventCL->takeOver(LastEvent);
+    ChipEventCL->setRecording();
+  }
```

Comment:
Could be expressed more simply.
```suggestion
  std::shared_ptr<chipstar::Event> LastEvent = getLastEvent();
  ChipEventCL->takeOver(LastEvent ? LastEvent : enqueueMarker());
  ChipEventCL->setRecording();
```

Reply:
nice.